### ISSUE TITLE
memleak failed to map address and printout <null sym>

### ISSUE THREAD

**Issue by gbh1hc** [iovisor/bcc#5165]
Hello everyone, I am trying to use [memleak.c](https://github.com/iovisor/bcc/tree/master/libbpf-tools/memleak.c) but get error to reach out the function name that leads to memory leak

My system:
aarch64 Linux localhost 6.1.57-android14

My test program `test_trace`:
```
void leak_func(void)
{
    int* x = malloc(99);
}

int main(void)
{
    leak_func();
    return 0;
}
```
The output:
```
# ./memleak -c ./test_trace -O ./libc.so
using page size: 4096
tracing kernel: false
child created with pid: 25052
libbpf: elf: 'valloc' is 0 in symtab for './libc.so': should not be 0 in a shared library
libbpf: elf: 'valloc' is 0 in symtab for './libc.so': should not be 0 in a shared library
libbpf: elf: 'pvalloc' is 0 in symtab for './libc.so': should not be 0 in a shared library
libbpf: elf: 'pvalloc' is 0 in symtab for './libc.so': should not be 0 in a shared library
Tracing outstanding memory allocs...  Hit Ctrl-C to end
received go event. executing child command
[4:55:58] Top 10 stacks with outstanding allocations:
8858370048 bytes in 1 allocations from stack
...
        6 [<0000007f78092988>] scudo_malloc+0x2c [/apex/com.android.runtime/lib64/bionic/libc.so]
        7 [<0000007f7808c610>] malloc+0x2c [/apex/com.android.runtime/lib64/bionic/libc.so]
        8 [<000000 7ffffff000>] <null sym>
        9 [<000000 60c9280110>] main.cfi+0x24 [test_trace]
        10 [<0000007f7809ad8c>] __libc_init+0x70 [/apex/com.android.runtime/lib64/bionic/libc.so]
        11 [<00000060c9280058>] __cfi_check_fail+0x0 [test_trace]
```
```
$ cat /proc/`pidof test_trace`/maps
7fe6dbb000-7fe6ddd000 rw-p 00000000 00:00 0                              [stack]
7ffffff000-8000000000 --xp 00000000 00:00 0                              [uprobes]
```

Any feedback is appreciated  🙏

**Comment by yonghong-song** [comment]
I suspect that the missing symbol is due to uprobe which attached to the leak_func(). Could you double check that?

I suspect the following kernel patch fixed the issue:
  https://lkml.kernel.org/r/20240729175223.23914-1-andrii@kernel.org
but it is only available at 6.12 and later.

**Comment by gbh1hc** [comment]
> I suspect that the missing symbol is due to uprobe which attached to the leak_func(). Could you double check that?
> 
> I suspect the following kernel patch fixed the issue: https://lkml.kernel.org/r/20240729175223.23914-1-andrii@kernel.org but it is only available at 6.12 and later.

Thanks for your feedback; it seems you are right
Checked address of <null_sym>
```bash
8 [<000000 7ffffff000>] <null sym>
```
```bash
cat /proc/test_trace/maps
7ffffff000-8000000000 --xp 00000000 00:00 0                              [uprobes]
```
But sometimes It can get the leaked function's name but sometimes it does not
Anyway, the problem of this issue is resolved. I close it then.
